libcontainer/cgroups/fs2/fs2.go

```diff
@@ -258,17 +260,105 @@ func (m *manager) Exists() bool {
 	return cgroups.PathExists(m.dirPath)
 }
 
-func (m *manager) CreateChildCgroup(container *configs.Config) error {
-	// sysbox-runc: implement this function
-	return fmt.Errorf("Systemd not supported")
+func (m *manager) CreateChildCgroup(config *configs.Config) error {
+
+	// Change the cgroup ownership to match the root user in the system
+	// container (needed for delegation).
+	path := m.dirPath
+
+	rootuid, err := config.HostRootUID()
+	if err != nil {
+		return err
+	}
+	rootgid, err := config.HostRootGID()
+	if err != nil {
+		return err
+	}
+
+	if err := os.Chown(path, rootuid, rootgid); err != nil {
+		return fmt.Errorf("Failed to change owner of cgroup %s", path)
+	}
+
+	// Change ownership of some of the files inside the sys container's cgroup;
+	// for cgroups v2 we only change the ownership of a subset of the files, as
+	// specified in section "Cgroups Delegation: Delegating a Hierarchy to a Less
+	// Privileged User" in cgroups(7).
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		fname := file.Name()
+
+		if fname == "cgroup.procs" ||
+			fname == "cgroup.subtree_control" ||
+			fname == "cgroup.threads" {
+
+			absFileName := filepath.Join(path, fname)
+			if err := os.Chown(absFileName, rootuid, rootgid); err != nil {
+				return fmt.Errorf("Failed to change owner for file %s", absFileName)
+			}
+		}
+	}
+
+	// Create a leaf cgroup to be used for the sys container's init process (and
```

Comment:
See some minor typos below, which also apply to the equivalent method in `unifiedManager` class.

Reply:
Got it ... fixed in both places.